src/SubjectFilter.php

```diff
@@ -102,10 +116,12 @@ class SubjectFilter
     public function __construct(
         ValidateSpec $validate_spec,
         SanitizeSpec $sanitize_spec,
+        SubSpecFactory $sub_spec_factory,
```

Comment:
I am little concerned about the change in signature for this.

I wonder whether this will make a BC break. Apart from this, I am ok with merging this PR.

Reply:
Oh this is 3.x, so no worries :) .